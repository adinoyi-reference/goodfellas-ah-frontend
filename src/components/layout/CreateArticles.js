import React from 'react';
import Header from '../shared/header';
import '../../styles/components/createArticles.scss';

const CreateArticles = () => (
  <div className="article">
    <Header />
    <hr />
    <div className="articles-card">
      <div className="article-buttons">
        <button className="btn article-whitebutton" type="submit">
            Publish
        </button>
        <button className="btn article-whitebutton" type="submit">
            Save for later
        </button>
      </div>
      <form>
        <input type="text" id="title" className="form" placeholder="Give me a title..." />
        <textarea name="body" id="content" placeholder="What would you like to talk about?" />
      </form>
      <div className="article-fixed-buttons">
        <button className="btn article-fixed-whitebutton" type="submit">
            Publish
        </button>
        <button className="btn article-fixed-whitebutton" type="submit">
            Save for later
        </button>
      </div>
    </div>
  </div>
);

export default CreateArticles;
