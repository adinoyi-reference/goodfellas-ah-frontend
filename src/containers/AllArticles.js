import React, { Component } from 'react';
import { connect } from 'react-redux';
import Card from './ArticleCard';
import RecentStory from '../components/article/RecentStory';
import SecondArticleCard from '../components/article/SecondArticleCard';
import { getArticles } from '../actions/articleActions';
import dice from '../assets/dice.png';
import '../styles/styles.scss';


class AllArticles extends Component {
  componentWillMount = () => {
    const { getArticles: getAllArticles } = this.props;

    getAllArticles();
  }


  render() {
    return (
      <div>
        <div className="feed">
        Stories From Your Feed
          <img src={dice} alt="" />
        </div>
        <div className="hero-lowerbody-articles">
          <Card />
        </div>
        <hr className="card-hr" />
        <div className="second-card">
          <div className="all-articles-card">
            <div className="buttons">
              <button type="button">All Articles</button>
              <button type="button">Favourites</button>
            </div>

            <SecondArticleCard />
            <SecondArticleCard />

          </div>
          <div className="recent-stories">
            <div className="recent-stories-box">
              <p className="title">Recent Stories</p>

              <RecentStory />
              <RecentStory />

            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  error: state.articles.error,
  articles: state.articles.articles,
});

export default connect(mapStateToProps, { getArticles })(AllArticles);
