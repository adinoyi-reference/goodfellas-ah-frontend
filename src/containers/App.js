import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import Hero from '../views/Hero';
import Signup from '../views/Signup';
import Signin from '../views/Signin';
import Profile from '../views/Profile';
import '../styles/styles.scss';

export default () => (
  <BrowserRouter>
    <div>
      <Route exact path="/" component={Hero} />
      <Route path="/auth/signin" render={(props) => <Signin formtype="signin" {...props} />} />
      <Route path="/auth/signup" render={(props) => <Signup formtype="signup" {...props} />} />
      <Route path="/user/profile" component={Profile} />
    </div>
  </BrowserRouter>
);
