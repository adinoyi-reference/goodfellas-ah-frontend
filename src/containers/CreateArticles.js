import React, { Component } from 'react';
import { connect } from 'react-redux';
import swal from 'sweetalert';
import Editor from 'react-medium-editor';
import 'medium-editor/dist/css/medium-editor.css';
import 'medium-editor/dist/css/themes/default.css';
import ImageUploader from '../components/articles/imageUpload';
import publishArticle from '../actions/publishArticle';
import '../styles/views/createArticles.scss';
import { Loader } from '../components/shared/Loading';


export class CreateArticles extends Component {
  state = {
    title: '',
    body: '',
    imageLoad: false,
    image: '',
  }

  componentDidUpdate(prevProps) {
    const { status, history, publishedArticle } = this.props;
    if (prevProps.status.success !== status.success && status.success) {
      swal('Good job!', 'Article Saved Successfully!', 'success');
      setTimeout(() => history.push(`/articles/${publishedArticle.article.slug}`), 3000);
    } else if (prevProps.status.error !== status.error && status.error) {
      swal('Error!', 'Something Went Wrong!', 'error');
    }
  }


  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  generateImageTag = (url) => `<img class="inline-image" src="${url}" />`

  imageUploaded = (url) => {
    const imageHtml = this.generateImageTag(url);
    const { image } = this.state;
    this.setState((prevState) => ({ body: prevState.body + imageHtml, imageLoad: false }));

    if (image.length < 1) {
      this.setState({
        image: url,
      });
    }
  }

  handleSubmit = (event) => {
    const { history } = this.props;
    event.preventDefault();
    const { title, body } = this.state;
    if (!title || title === '<p><br></p>') return swal('Please add a title', 'Your article needs to have a title', 'warning');
    if (!body || body === '<p><br></p>') { return swal('Please add a body', 'The body of your article cannot be empty!', 'warning'); }
    const articlePayload = {
      ...this.state,
      description: body.split(' ').slice(0, 10).join(' '),
    };
    return history.push({
      pathname: '/tags',
      state: { articlePayload },
    });
  }

  render() {
    const handleEditorChange = (text, key) => this.setState({ [key]: text });
    const { imageUploadStatus } = this.props;
    const { title, body } = this.state;
    return (
      <div className="article-body">
        <div className="articles-card">
          <div className="article-buttons">
            <button className="btn article-whitebutton" type="submit" onClick={this.handleSubmit}>
            Publish
            </button>
          </div>
          <ImageUploader imageUploaded={this.imageUploaded} />

          <form>
            <Editor
              name="title"
              id="title"
              data-placeholder="Title"
              text={title}
              onChange={(e) => handleEditorChange(e, 'title')}
              options={{
                toolbar: {
                  buttons: ['bold', 'italic', 'underline', 'strikethrough', 'quote', 'anchor', 'h2', 'h3', 'orderedlist'],
                },
              }}
            />
            {imageUploadStatus.loading ? <Loader /> : (
              <Editor
                name="body"
                className="editable"
                data-placeholder="What would you like to talk about?"
                text={body}
                onChange={(e) => handleEditorChange(e, 'body')}
                options={{
                  toolbar: {
                    buttons: ['bold', 'italic', 'underline', 'strikethrough', 'quote', 'anchor', 'h2', 'h3', 'orderedlist'],
                  },
                }}
              />
            )}
          </form>

        </div>
        <div className="article-fixed-buttons">
          <button className="btn article-fixed-whitebutton" type="submit" onClick={this.handleSubmit}>
            Publish
          </button>
        </div>
      </div>
    );
  }
}


const mapStateToProps = ({
  imageUploadReducer: { status: imageUploadStatus },
  publishArticleReducer: { status, publishedArticle },
}) => ({
  status,
  publishedArticle,
  imageUploadStatus,
});

export default connect(mapStateToProps, { publishArticle })(CreateArticles);
