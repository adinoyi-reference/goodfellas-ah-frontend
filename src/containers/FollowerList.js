import React, { Fragment } from 'react';
import FollowCard from '../components/shared/FollowCard';
import Loading from '../components/shared/Loading';

const FollowerList = (props) => {
  if (props.loading) return <Loading />;
  return (
    <Fragment>
      {
        Array(5).fill().map((x, i) => <FollowCard key={i} type="follower" />)
      }
    </Fragment>);
};

export default FollowerList;
