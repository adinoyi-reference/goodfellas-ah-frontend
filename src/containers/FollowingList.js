import React, { Fragment } from 'react';
import FollowCard from '../components/shared/FollowCard';

const FollowingList = (props) => (
  <Fragment>
    {
      props.following.map((following) => <FollowCard key={following.id} type="following" user={following.followedUser} />)
    }
  </Fragment>);

export default FollowingList;
