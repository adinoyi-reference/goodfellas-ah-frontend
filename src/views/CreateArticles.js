import React, { Component } from 'react';
import { connect } from 'react-redux';
import swal from 'sweetalert';
import Header from '../components/shared/header';
import publishArticle from '../actions/publishArticle';
import '../styles/components/createArticles.scss';


class CreateArticles extends Component {
  state = {
    title: '',
    body: '',
  }

  componentDidUpdate() {
    if (this.props.status.success) {
      swal('Good job!', 'Article Published Successfully!', 'success');
    } else if (this.props.status.error) {
      swal('Error!', 'Something Went Wrong!', 'error');
    }
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { body } = this.state;
    const articlePayload = {
      ...this.state,
      description: body.split(' ').slice(0, 3).join(' '),
    };
    this.props.publishArticle(articlePayload);
  }


  render() {
    return (
      <div className="article">
        <Header />
        <hr />
        <div className="articles-card">
          <div className="article-buttons">
            <button className="btn article-whitebutton" type="submit" onClick={this.handleSubmit}>
            Publish
            </button>
            <button className="btn article-whitebutton" type="submit">
            Save for later
            </button>
          </div>
          <div className="fab" id="masterfab">+</div>
          <form>
            <input name="title" type="text" id="title" className="form" placeholder="Give me a title..." onChange={this.handleChange} />
            <textarea name="body" id="content" placeholder="What would you like to talk about?" onChange={this.handleChange} />
          </form>

        </div>
        <div className="article-fixed-buttons">
          <button className="btn article-fixed-whitebutton" type="submit">
            Save for later
          </button>
          <button className="btn article-fixed-whitebutton" type="submit" onClick={this.handleSubmit}>
            Publish
          </button>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ publishArticleReducer: { status, publishedArticle } }) => ({
  status,
  publishedArticle,
});

export default connect(mapStateToProps, { publishArticle })(CreateArticles);
