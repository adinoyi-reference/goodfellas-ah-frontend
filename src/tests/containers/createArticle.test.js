import React from 'react';
import toJson from 'enzyme-to-json';
import { CreateArticles } from '../../containers/CreateArticles';

describe('create artcicle container test suite', () => {
  const props = {
    status: {
      error: false,
      success: false,
    },
    publishedArticle: {},
    imageUploadStatus: {
      error: false,
      success: false,
    },
    publishArticle: jest.fn(),
  };
  const wrapper = shallow(<CreateArticles {...props} />);


  it('renders properly when mounted', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not dispatch publishArticle when the publish button is clicked without title or body', () => {
    const publishButton = wrapper.find('.article-whitebutton').at(0);
    publishButton.simulate('click', { preventDefault: jest.fn() });
    expect(props.publishArticle).not.toBeCalled();
  });
  it('should dispatch publishArticle when the publish button is clicked with title and body', () => {
    const publishButton = wrapper.find('.article-whitebutton').at(0);
    wrapper.setState({ title: 'some title', body: 'somebody' });
    publishButton.simulate('click', { preventDefault: jest.fn() });
    expect(props.publishArticle).toBeCalled();
    props.publishArticle.mockRestore();
  });
  describe('render', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapper);
      expect(tree).toMatchSnapshot();
    });
  });
  it('should handle upload', () => {
    const instance = wrapper.instance();
    instance.imageUploaded('http://dummy-url');
    instance.forceUpdate();
    wrapper.update();
  });
  it('should handle submission errors', () => {
    const preventDefault = jest.fn();
    const btn = wrapper.find('.article-fixed-buttons > button');
    btn.simulate('click', {
      preventDefault,
    });
    expect(preventDefault).toBeCalled();
    const instance = wrapper.instance();
    instance.setState({
      title: 'valid title',
      body: null,
    });
    instance.forceUpdate();
    btn.simulate('click', {
      preventDefault: jest.fn(),
    });
    expect(preventDefault).toBeCalled();
  });
  it('should handle change', () => {
    const instance = wrapper.instance();
    instance.handleChange({
      target: {
        name: 'body',
        value: 'test',
      },
    });
    instance.forceUpdate();
    expect(instance.state.body).toEqual('test');
  });
  it('should call componentDidUpdate', () => {
    const props1 = {
      status: {
        error: false,
        success: false,
      },
      publishedArticle: {},
      imageUploadStatus: {
        error: false,
        success: false,
      },
      publishArticle: jest.fn(),
    };
    const wrapper1 = shallow(<CreateArticles {...props1} />);
    const newProps = {
      history: {
        push: jest.fn(),
      },
      publishedArticle: {
        article: {
          slug: 'dummy-slug',
        },
      },
      status: {
        success: false,
        error: true,
      },
    };
    wrapper1.setProps(newProps);
    wrapper1.update();
    expect(newProps.history.push).not.toBeCalled();
  });
  it('should call componentDidUpdate', () => {
    jest.useFakeTimers();
    const props1 = {
      status: {
        error: false,
        success: false,
      },
      publishedArticle: {},
      imageUploadStatus: {
        error: false,
        success: false,
      },
      publishArticle: jest.fn(),
    };
    const wrapper1 = shallow(<CreateArticles {...props1} />);
    const newProps = {
      history: {
        push: jest.fn(),
      },
      publishedArticle: {
        article: {
          slug: 'dummy-slug',
        },
      },
      status: {
        success: true,
        error: false,
      },
    };
    wrapper1.setProps(newProps);
    wrapper1.update();

    jest.runAllTimers();
    const value = `/articles/${newProps.publishedArticle.article.slug}`;
    expect(newProps.history.push).toBeCalled();
    expect(newProps.history.push).toBeCalledWith(value);
  });
});
