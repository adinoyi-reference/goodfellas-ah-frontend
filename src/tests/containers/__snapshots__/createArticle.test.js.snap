// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create artcicle component test suite renders properly when mounted 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateArticles
    imageUploadStatus={
      Object {
        "error": false,
        "success": false,
      }
    }
    publishArticle={[MockFunction]}
    publishedArticle={Object {}}
    status={
      Object {
        "error": false,
        "success": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="articles-card"
        >
          <div
            className="article-buttons"
          >
            <button
              className="btn article-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Publish
            </button>
            <button
              className="btn article-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Save for later
            </button>
          </div>
          <Connect(ImageUploader)
            imageUploaded={[Function]}
          />
          <form>
            <ReactMediumEditor
              data-placeholder="Title"
              id="title"
              name="title"
              onChange={[Function]}
              options={
                Object {
                  "toolbar": Object {
                    "buttons": Array [
                      "bold",
                      "italic",
                      "underline",
                      "strikethrough",
                      "quote",
                      "anchor",
                      "h2",
                      "h3",
                      "orderedlist",
                    ],
                  },
                }
              }
              tag="div"
              text=""
            />
            <ReactMediumEditor
              className="editable"
              data-placeholder="What would you like to talk about?"
              name="body"
              onChange={[Function]}
              options={
                Object {
                  "toolbar": Object {
                    "buttons": Array [
                      "bold",
                      "italic",
                      "underline",
                      "strikethrough",
                      "quote",
                      "anchor",
                      "h2",
                      "h3",
                      "orderedlist",
                    ],
                  },
                }
              }
              tag="div"
              text=""
            />
          </form>
        </div>,
        <div
          className="article-fixed-buttons"
        >
          <button
            className="btn article-fixed-whitebutton"
            onClick={[Function]}
            type="submit"
          >
            Save for later
          </button>
          <button
            className="btn article-fixed-whitebutton"
            onClick={[Function]}
            type="submit"
          >
            Publish
          </button>
        </div>,
      ],
      "className": "article-body",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="article-buttons"
            >
              <button
                className="btn article-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Publish
              </button>
              <button
                className="btn article-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Save for later
              </button>
            </div>,
            <Connect(ImageUploader)
              imageUploaded={[Function]}
            />,
            <form>
              <ReactMediumEditor
                data-placeholder="Title"
                id="title"
                name="title"
                onChange={[Function]}
                options={
                  Object {
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  }
                }
                tag="div"
                text=""
              />
              <ReactMediumEditor
                className="editable"
                data-placeholder="What would you like to talk about?"
                name="body"
                onChange={[Function]}
                options={
                  Object {
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  }
                }
                tag="div"
                text=""
              />
            </form>,
          ],
          "className": "articles-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn article-whitebutton"
                  onClick={[Function]}
                  type="submit"
                >
                  Publish
                </button>,
                <button
                  className="btn article-whitebutton"
                  onClick={[Function]}
                  type="submit"
                >
                  Save for later
                </button>,
              ],
              "className": "article-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Publish",
                  "className": "btn article-whitebutton",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Publish",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Save for later",
                  "className": "btn article-whitebutton",
                  "onClick": [Function],
                  "type": "submit",
                },
                "ref": null,
                "rendered": "Save for later",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "imageUploaded": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <ReactMediumEditor
                  data-placeholder="Title"
                  id="title"
                  name="title"
                  onChange={[Function]}
                  options={
                    Object {
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    }
                  }
                  tag="div"
                  text=""
                />,
                <ReactMediumEditor
                  className="editable"
                  data-placeholder="What would you like to talk about?"
                  name="body"
                  onChange={[Function]}
                  options={
                    Object {
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    }
                  }
                  tag="div"
                  text=""
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "data-placeholder": "Title",
                  "id": "title",
                  "name": "title",
                  "onChange": [Function],
                  "options": Object {
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  },
                  "tag": "div",
                  "text": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "editable",
                  "data-placeholder": "What would you like to talk about?",
                  "name": "body",
                  "onChange": [Function],
                  "options": Object {
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  },
                  "tag": "div",
                  "text": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn article-fixed-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Save for later
            </button>,
            <button
              className="btn article-fixed-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Publish
            </button>,
          ],
          "className": "article-fixed-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save for later",
              "className": "btn article-fixed-whitebutton",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Save for later",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Publish",
              "className": "btn article-fixed-whitebutton",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Publish",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="articles-card"
          >
            <div
              className="article-buttons"
            >
              <button
                className="btn article-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Publish
              </button>
              <button
                className="btn article-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Save for later
              </button>
            </div>
            <Connect(ImageUploader)
              imageUploaded={[Function]}
            />
            <form>
              <ReactMediumEditor
                data-placeholder="Title"
                id="title"
                name="title"
                onChange={[Function]}
                options={
                  Object {
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  }
                }
                tag="div"
                text=""
              />
              <ReactMediumEditor
                className="editable"
                data-placeholder="What would you like to talk about?"
                name="body"
                onChange={[Function]}
                options={
                  Object {
                    "toolbar": Object {
                      "buttons": Array [
                        "bold",
                        "italic",
                        "underline",
                        "strikethrough",
                        "quote",
                        "anchor",
                        "h2",
                        "h3",
                        "orderedlist",
                      ],
                    },
                  }
                }
                tag="div"
                text=""
              />
            </form>
          </div>,
          <div
            className="article-fixed-buttons"
          >
            <button
              className="btn article-fixed-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Save for later
            </button>
            <button
              className="btn article-fixed-whitebutton"
              onClick={[Function]}
              type="submit"
            >
              Publish
            </button>
          </div>,
        ],
        "className": "article-body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="article-buttons"
              >
                <button
                  className="btn article-whitebutton"
                  onClick={[Function]}
                  type="submit"
                >
                  Publish
                </button>
                <button
                  className="btn article-whitebutton"
                  onClick={[Function]}
                  type="submit"
                >
                  Save for later
                </button>
              </div>,
              <Connect(ImageUploader)
                imageUploaded={[Function]}
              />,
              <form>
                <ReactMediumEditor
                  data-placeholder="Title"
                  id="title"
                  name="title"
                  onChange={[Function]}
                  options={
                    Object {
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    }
                  }
                  tag="div"
                  text=""
                />
                <ReactMediumEditor
                  className="editable"
                  data-placeholder="What would you like to talk about?"
                  name="body"
                  onChange={[Function]}
                  options={
                    Object {
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    }
                  }
                  tag="div"
                  text=""
                />
              </form>,
            ],
            "className": "articles-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn article-whitebutton"
                    onClick={[Function]}
                    type="submit"
                  >
                    Publish
                  </button>,
                  <button
                    className="btn article-whitebutton"
                    onClick={[Function]}
                    type="submit"
                  >
                    Save for later
                  </button>,
                ],
                "className": "article-buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Publish",
                    "className": "btn article-whitebutton",
                    "onClick": [Function],
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Publish",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Save for later",
                    "className": "btn article-whitebutton",
                    "onClick": [Function],
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": "Save for later",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "imageUploaded": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <ReactMediumEditor
                    data-placeholder="Title"
                    id="title"
                    name="title"
                    onChange={[Function]}
                    options={
                      Object {
                        "toolbar": Object {
                          "buttons": Array [
                            "bold",
                            "italic",
                            "underline",
                            "strikethrough",
                            "quote",
                            "anchor",
                            "h2",
                            "h3",
                            "orderedlist",
                          ],
                        },
                      }
                    }
                    tag="div"
                    text=""
                  />,
                  <ReactMediumEditor
                    className="editable"
                    data-placeholder="What would you like to talk about?"
                    name="body"
                    onChange={[Function]}
                    options={
                      Object {
                        "toolbar": Object {
                          "buttons": Array [
                            "bold",
                            "italic",
                            "underline",
                            "strikethrough",
                            "quote",
                            "anchor",
                            "h2",
                            "h3",
                            "orderedlist",
                          ],
                        },
                      }
                    }
                    tag="div"
                    text=""
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "data-placeholder": "Title",
                    "id": "title",
                    "name": "title",
                    "onChange": [Function],
                    "options": Object {
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    },
                    "tag": "div",
                    "text": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "editable",
                    "data-placeholder": "What would you like to talk about?",
                    "name": "body",
                    "onChange": [Function],
                    "options": Object {
                      "toolbar": Object {
                        "buttons": Array [
                          "bold",
                          "italic",
                          "underline",
                          "strikethrough",
                          "quote",
                          "anchor",
                          "h2",
                          "h3",
                          "orderedlist",
                        ],
                      },
                    },
                    "tag": "div",
                    "text": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "form",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn article-fixed-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Save for later
              </button>,
              <button
                className="btn article-fixed-whitebutton"
                onClick={[Function]}
                type="submit"
              >
                Publish
              </button>,
            ],
            "className": "article-fixed-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save for later",
                "className": "btn article-fixed-whitebutton",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Save for later",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Publish",
                "className": "btn article-fixed-whitebutton",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Publish",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
